docs: improve runtimeConfig example (#5785)
nuxt/framework: docs/content/2.guide/2.features/10.runtime-config.md

```ts [nuxt.config.ts]
export default defineNuxtConfig({
  runtimeConfig: {

>     apiSecret: process.env.NUXT_API_SECRET || '', // .env value + an empty string as a fallback value

Reviewer: ```suggestion
    apiSecret: '', // can be overridden by NUXT_API_SECRET environment variable
```

    public: {

>       apiBase: process.env.NUXT_PUBLIC_API_BASE || 'someBase', // Or .env value + a fallback value
>       apiPublicKey: 'somePublicApiKey', // Or a default value

Reviewer: ```suggestion
      apiBase: '', // can be overridden by NUXT_PUBLIC_API_BASE environment variable
```
